pyquil/quil.py

```diff
@@ -454,94 +457,19 @@ def dagger(self, inv_dict=None, suffix="-INV"):
-                    used_indexes.add(instr.qubit.index)
-
-        def find_available_index():
-            # Just do a linear search.
-            for i in count(start=0, step=1):
-                if i not in used_indexes:
-                    return i
-
-        qubit_mapping = dict()
-
-        def remap_qubit(qubit):
-            if not isinstance(qubit, QubitPlaceholder):
-                return qubit
-            if id(qubit) in qubit_mapping:
-                return qubit_mapping[id(qubit)]
-            else:
-                available_index = find_available_index()
-                used_indexes.add(available_index)
-                remapped_qubit = Qubit(available_index)
-                qubit_mapping[id(qubit)] = remapped_qubit
-                return remapped_qubit
-
-        label_mapping = dict()
-        label_counter = 1
-
-        def remap_label(placeholder):
-            if id(placeholder) in label_mapping:
-                return label_mapping[id(placeholder)]
-            else:
-                label = Label(placeholder.prefix + str(label_counter))
-                label_mapping[id(placeholder)] = label
-                return label
-
-        result = []
-        for instr in self._instructions:
-            # Remap qubits on Gate and Measurement instructions
-            if isinstance(instr, Gate):
-                remapped_qubits = [remap_qubit(q) for q in instr.qubits]
-                result.append(Gate(instr.name, instr.params, remapped_qubits))
-            elif isinstance(instr, Measurement):
-                result.append(Measurement(remap_qubit(instr.qubit), instr.classical_reg))
-
-            # Remap any label placeholders on jump or target instructions
-            elif isinstance(instr, Jump) and isinstance(instr.target, LabelPlaceholder):
-                result.append(Jump(remap_label(instr.target)))
-                label_counter += 1
-            elif isinstance(instr, JumpTarget) and isinstance(instr.label, LabelPlaceholder):
-                result.append(JumpTarget(remap_label(instr.label)))
-                label_counter += 1
-            elif isinstance(instr, JumpConditional) and isinstance(instr.target, LabelPlaceholder):
-                new_label = remap_label(instr.target)
-                if isinstance(instr, JumpWhen):
-                    result.append(JumpWhen(new_label, instr.condition))
-                elif isinstance(instr, JumpUnless):
-                    result.append(JumpUnless(new_label, instr.condition))
-                else:
-                    raise TypeError("Encountered a JumpConditional that wasn't JumpWhen or JumpUnless: {} {}"
-                                    .format(type(instr), instr))
-                label_counter += 1
-
-            # Otherwise simply add it to the result
-            else:
-                result.append(instr)
-
-        return result
+        self._synthesized_instructions = instantiate_labels(self._instructions)
+        return self
```

Comment:
Why is there a need to modify the return type of this and also document the changes? It's a private method

Reply:
was rooting around in this code anyways and it made more sense for a method named `synthesize` to do mutation. Documented the change in case anyone was curious